dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/NeuralNetwork.java

```diff
@@ -172,7 +176,9 @@ public void train(final INDArray input, final int label) {
    */
   public INDArray[] backPropagate(final INDArray[] activations,
                                   final INDArray label) {
-    return backPropagateTo(0, activations, label);
+    // Process backpropagation to the second layer
+    // because the error the first layer generates, is not needed to generate a gradient for learning the first layer.
```

Comment:
I'm still not satisfied with this explanation. This makes it sound like the first layer is always not learnable. Also, the words "the error the first layer generates" can be misunderstood if the reader doesn't know that weight updates are done by the next layer's back propagation call.